internal/request/graphql/schema/collection.go

```diff
@@ -596,6 +624,92 @@ func policyFromAST(directive *ast.Directive) (client.PolicyDescription, error) {
 	return policyDesc, nil
 }
 
+func embeddingFromAST(directive *ast.Directive) (*client.EmbeddingDescription, error) {
+	embedding := &client.EmbeddingDescription{}
+	for _, arg := range directive.Arguments {
+		switch arg.Name.Value {
+		case types.EmbeddingDirectivePropFields:
+			val, ok := arg.Value.(*ast.ListValue)
+			if !ok {
+				return nil,
+					NewErrEmbeddingInvalidProp[[]string](types.ConstraintsDirectivePropSize, arg.Value.GetValue())
+			}
+			fields := make([]string, len(val.Values))
+			for i, untypedField := range val.Values {
+				field, ok := untypedField.(*ast.StringValue)
+				if !ok {
+					return nil,
+						NewErrEmbeddingInvalidProp[[]string](types.ConstraintsDirectivePropSize, field.GetValue())
+				}
+				fields[i] = field.Value
+			}
+			embedding.Fields = fields
+		case types.EmbeddingDirectivePropModel:
+			val, ok := arg.Value.(*ast.StringValue)
+			if !ok {
+				return nil,
+					NewErrEmbeddingInvalidProp[string](types.EmbeddingDirectivePropModel, arg.Value.GetValue())
+			}
+			embedding.Model = val.Value
+		case types.EmbeddingDirectivePropProvider:
+			val, ok := arg.Value.(*ast.StringValue)
+			if !ok {
+				return nil,
+					NewErrEmbeddingInvalidProp[string](types.EmbeddingDirectivePropProvider, arg.Value.GetValue())
+			}
+			embedding.Provider = val.Value
+		case types.EmbeddingDirectivePropTemplate:
+			val, ok := arg.Value.(*ast.StringValue)
+			if !ok {
+				return nil,
+					NewErrEmbeddingInvalidProp[string](types.EmbeddingDirectivePropTemplate, arg.Value.GetValue())
+			}
+			embedding.Template = val.Value
+		case types.EmbeddingDirectivePropURL:
+			val, ok := arg.Value.(*ast.StringValue)
+			if !ok {
+				return nil,
+					NewErrEmbeddingInvalidProp[string](types.EmbeddingDirectivePropURL, arg.Value.GetValue())
+			}
+			embedding.URL = val.Value
+		default:
+			return nil,
+				NewErrDirectiveWithUnknownArg(types.ConstraintsDirectiveLabel, arg.Name.Value)
+		}
+	}
+	return embedding, nil
+}
+
+type constraintDescription struct {
+	Size int
+}
+
+func contraintsFromAST(kind client.FieldKind, directive *ast.Directive) (constraintDescription, error) {
+	constraints := constraintDescription{}
+	for _, arg := range directive.Arguments {
+		switch arg.Name.Value {
+		case types.ConstraintsDirectivePropSize:
+			if !kind.IsArray() {
+				return constraintDescription{}, errors.New("size constraint can only be applied to array fields")
```

Comment:
todo: This error should be declared in an errors.go file

Reply:
Done